File: 168. Excel Sheet Column Title/excelsheetcolumntitle.py
class Solution:
    def convertToTitle(self, columnNumber: int) -> str:
        result = []
        i = 0

        while columnNumber > 0:
            columnNumber -= 1
            result.append(chr(ord('A') + (columnNumber % 26)))
            i += 1
            columnNumber //= 26

        l, r = 0, i - 1

        while l < r:
            result[l], result[r] = result[r], result[l]
            l += 1
            r -= 1

        return ''.join(result)

File: 168. Excel Sheet Column Title/excelsheetcolumntitle.java
class Solution {
    public String convertToTitle(int columnNumber) {
        char[] result = new char[8];
        int i = 0;

        while (columnNumber > 0) {
            columnNumber--;
            result[i++] = (char) ('A' + (columnNumber % 26));
            columnNumber /= 26;
        }

        for (int l = 0, r = i - 1; l < r; l++, r--) {
            char temp = result[l];
            result[l] = result[r];
            result[r] = temp;
        }

        return new String(result, 0, i);
    }
}
